src/mono/wasm/runtime/loader/exit.ts

```diff
@@ -1,10 +1,34 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-import { ENVIRONMENT_IS_NODE, ENVIRONMENT_IS_SHELL, ENVIRONMENT_IS_WEB, INTERNAL, loaderHelpers, runtimeHelpers } from "./globals";
-import { mono_log_debug, consoleWebSocket, mono_log_error, mono_log_info_no_prefix } from "./logging";
+import { ENVIRONMENT_IS_NODE, ENVIRONMENT_IS_SHELL, ENVIRONMENT_IS_WEB, INTERNAL, loaderHelpers, mono_assert, runtimeHelpers } from "./globals";
+import { mono_log_debug, consoleWebSocket, mono_log_error, mono_log_info_no_prefix, mono_log_warn } from "./logging";
+
+
+let ABORT = false;
+let EXIT = false;
+let EXIT_CODE: number | undefined = undefined;
```

Comment:
I'm  unclear on what is supposed to happen if you call `mono_exit` on a worker.  There is nothing that propagates values of `ABORT`, `EXIT`, and `EXIT_CODE` between workers. So if a worker calls `mono_exit` it will always succeed.  And meanwhile other workers and the browser thread will still see `loaderHelpers.is_runtime_running() === true`.

Would it make sense to store these flags in the shared array buffer?

Also is this `loaderHelpers` the right module for this? not `runtimeHelpers`? I understand you want to be able to abort startup, but it seems like once everythign is running, this should be managed by the (worker-aware) runtime module instead.

Reply:
both `loaderHelpers` and `runtimeHelpers` are present on threads because both JS modules are loaded.

Reply:
about MT, when wasm code is already loaded, we have `mono_wasm_exit` and use it to propagate the event. You are right that JS state on web workers would be stale. But on the other hand, emscripten will terminate all workers so it doesn't really mater.

Reply:
> My point is that `loaderHelpers` is loaded before the web assembly instance is created, so it doesn't necessarily have access to the memory buffer, right?

Right